examples/platforms/nrf528xx/src/radio.c

```diff
@@ -489,7 +491,16 @@ otError otPlatRadioReceive(otInstance *aInstance, uint8_t aChannel)
 
     nrf_802154_tx_power_set(GetTransmitPowerForChannel(aChannel));
 
-    result = nrf_802154_receive();
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+    if (sCslSampleDur > 0)
```

Comment:
I would suggest to use a separate API like `otPlatRadioReceiveAt` for this purpose rather than reuse `otPlatRadioReceive` with some global parameters. There are some wakes not related to CSL managed by OT stack. If we change this implementation like this, it would mess up the OT managed wake.

Reply:
Makes sense, I'll update.